corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -22,3 +28,27 @@ def force_update_cleanliness_flags():
     # that there are no bugs in the weekly task.
     # If we haven't seen any issues by the end of 2015 (so 6 runs) we should remove this.
     set_cleanliness_flags_for_all_domains(force_full=True)
+
+
+@task(queue=ASYNC_RESTORE_QUEUE)
+def get_async_restore_payload(restore_config):
+    """Process an async restore
+    """
+    response = restore_config.generate_payload(async_task=current_task)
```

Comment:
I'm not sure to be honest. From the celery docs:
> `current_task`: The task currently being executed (only set in the worker, or when eager/apply is used).
> `bind`: The bind argument means that the function will be a “bound method” so that you can access attributes and methods on the task type instance.

I think `current_task` is probably better suited to being used outside of the task itself (i.e importing `current_task` in a different module), whereas `bind` is probably better for inside. Since I'm passing the task into the method I'm calling, it might have been more suitable to use `bind`.